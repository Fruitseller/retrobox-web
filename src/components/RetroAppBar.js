import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import Button from 'material-ui/Button';

const styles = theme => ({
  root: {
    marginTop: theme.spacing.unit * 3,
    width: '100%'
  },
  flex: {
    flex: 1
  }
});

const RetroAppBar = props => {
  return (
    <div className={props.classes.root}>
      <AppBar>
        <Toolbar>
          <Typography
            type="title"
            color="inherit"
            className={props.classes.flex}
          >
            Retrobox
          </Typography>
          <Button color="contrast" onClick={props.handleOnClick}>
            {props.buttonText}
          </Button>
        </Toolbar>
      </AppBar>
    </div>
  );
};

RetroAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
  buttonText: PropTypes.string.isRequired,
  handleOnClick: PropTypes.func.isRequired
};

export default withStyles(styles)(RetroAppBar);
