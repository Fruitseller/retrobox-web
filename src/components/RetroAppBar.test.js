import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import RetroAppBar from './RetroAppBar';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const retroAppBar = renderer.create(
    <RetroAppBar buttonText="This is a buttonText" handleOnClick={() => {}} />
  );
  expect(retroAppBar).toMatchSnapshot();
});
