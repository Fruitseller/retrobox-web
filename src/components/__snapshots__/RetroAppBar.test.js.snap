// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="RetroAppBar-root-1"
>
  <header
    className="MuiPaper-root-10 MuiPaper-shadow4-16 MuiAppBar-root-3 MuiAppBar-positionFixed-4 MuiAppBar-colorPrimary-8 mui-fixed"
  >
    <div
      className="MuiToolbar-root-37 MuiToolbar-gutters-38"
    >
      <h2
        className="MuiTypography-root-39 MuiTypography-title-45 MuiTypography-colorInherit-58 RetroAppBar-flex-2"
      >
        Retrobox
      </h2>
      <button
        className="MuiButtonBase-root-77 MuiButton-root-63 MuiButton-flatContrast-68"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label-65"
        >
          This is a buttonText
        </span>
        <span
          className="MuiTouchRipple-root-79"
        />
      </button>
    </div>
  </header>
</div>
`;
